apps/ledger-live-mobile/src/screens/Platform/v2/Catalog/index.tsx

```diff
@@ -0,0 +1,187 @@
+
+  const {
+    input,
+    inputRef,
+    onChange,
+    onCancel,
+    onFocus,
+    isActive,
+    isSearching,
+    result,
+  } = useSearch({
+    list: manifests,
+    options: {
+      keys: ["name"],
+    },
+  });
+
+  // TODO: Move inside the custom hook
+  useEffect(() => {
+    isActive && setSelected(initialSelectedState);
+  }, [isActive, setSelected, initialSelectedState]);
+
+  const {
+    isDismissed,
+    isReadOnly,
+    openApp,
+    prompt,
+    name,
+    icon,
+    isOpened,
+    onClose,
+    isChecked,
+    toggleCheck,
+    onContinue,
+  } = useDisclaimer(pushRecentlyUsed);
+
+  useDeeplinkEffect(manifests, openApp);
+
+  const onSelect = useCallback(
+    (manifest: LiveAppManifest) => {
+      if (!isDismissed && !isReadOnly && manifest.author !== "ledger") {
+        prompt(manifest);
+      } else {
+        openApp(manifest);
+      }
+    },
+    [isDismissed, isReadOnly, openApp, prompt],
+  );
+
+  return (
+    <TabBarSafeAreaView edges={["bottom", "left", "right"]}>
+      {/* TODO: put under the animation header and style  */}
+      <TrackScreen category="Platform" name="Catalog" />
+      <DAppDisclaimer
+        name={name}
+        icon={icon}
+        isOpened={isOpened}
+        onClose={onClose}
+        isChecked={isChecked}
+        toggleCheck={toggleCheck}
+        onContinue={onContinue}
+      />
+
+      {isActive ? (
+        <Search
+          manifests={manifests}
+          recentlyUsed={recentlyUsed}
+          title={title}
+          input={input}
+          inputRef={inputRef}
+          onCancel={onCancel}
+          onChange={onChange}
+          onFocus={onFocus}
+          result={result}
+          isSearching={isSearching}
+          isActive={isActive}
+          onSelect={onSelect}
+        />
+      ) : (
+        <>
+          <Layout
+            title={title}
+            listStickyElement={[3]}
+            topHeaderContent={
+              <TouchableOpacity
+                hitSlop={{
+                  bottom: 10,
+                  left: 24,
+                  right: 24,
+                  top: 10,
+                }}
+                style={{ paddingVertical: 16 }}
+                onPress={navigation.goBack}
+              >
+                <ArrowLeft size={18} color={colors.neutral.c100} />
+              </TouchableOpacity>
+            }
+            titleHeaderContent={
+              <Flex marginBottom={16}>
+                <Text fontWeight="bold" variant="h1Inter">
+                  {title}
+                </Text>
+                <Text variant="large">{t("browseWeb3.catalog.subtitle")}</Text>
+              </Flex>
+            }
+            middleHeaderContent={
+              <SearchBar
+                input={input}
+                inputRef={inputRef}
+                onChange={onChange}
+                onCancel={onCancel}
+                onFocus={onFocus}
+                isActive={isActive}
+              />
```

Comment:
This can be just a dummy search bar. When user clicks, `<Search />` would be rendered instead so no need to pass these props except `onFocus`.